src/rust/ide/src/controller/project.rs

```diff
@@ -98,9 +101,12 @@ impl ExecutionContextsRegistry {
 
 /// Project controller's state.
 #[allow(missing_docs)]
-#[derive(Clone,CloneRef,Debug)]
+#[derive(Clone,CloneRef,Derivative)]
+#[derivative(Debug)]
 pub struct Handle {
-    pub project_name        : Rc<String>,
+    pub project_metadata    : Rc<RefCell<ProjectMetadata>>,
```

Comment:
I think that `ProjectMetadata` is not the type we want to store here — it represents the project entry in the list. In particular, `lastOpened` will be most certainly outdated after we open the project. 

***

Also, this might be a good opportunity to split off the project model from the project controller, to achieve same model as we have for e.g. execution context. 
The model should contain mutable state that corresponds to some remote entity, while controller should be immutable with regards to remote operations, it is just a set of data that we can use to call the remote.
The refcell itself should also be the implementation detail of that model type.

@farmaazon will likely be able to explain this better and give advice whether this is something we want pursue soon-ish for project.

Reply:
As discussed on Discord, we will leave this refactoring for later since it needs to be planned as a separated task.

Reply:
I introduced a new struct called `ProjectData` inside `controller/project.rs`. The MVC refactoring is ready in the project name view PR.